src/v2/coins/eth.ts

```diff
@@ -805,6 +812,182 @@ class Eth extends BaseCoin {
+      const walletPassphrase = buildParams.walletPassphrase;
+
+      const userKeychain = yield this.keychains().get({ id: wallet._wallet.keys[0] });
+      const userPrv = wallet.getUserPrv({ keychain: userKeychain, walletPassphrase });
+      const userPrvBuffer = utxoLib.HDNode.fromBase58(userPrv).getKey().getPrivateKeyBuffer();
+      if (!recipients || !Array.isArray(recipients)) {
+        throw new Error('expecting array of recipients');
+      }
+
+      // Right now we only support 1 recipient
+      if (recipients.length !== 1) {
+        throw new Error('must send to exactly 1 recipient');
+      }
+      const recipientAddress = recipients[0].address;
+      const recipientAmount = recipients[0].amount;
+      const feeEstimateParams = {
+        recipient: recipientAddress,
+        hop: true,
+      };
+      const feeEstimate = yield this.feeEstimate(feeEstimateParams);
+      const gasLimit = feeEstimate.gasLimitEstimate;
+      // shift by 9 because result is in gwei and we want wei
+      const gasPrice = Math.round((feeEstimate.feeEstimate) / gasLimit);
+      const gasPriceMax = gasPrice * 5;
+      // Payment id as  current datetime so its always unique every time we build one
+      const paymentId = (new Date().toISOString());
+      const hopDigest: Buffer = Eth.getHopDigest([recipientAddress, recipientAmount, gasPriceMax, gasLimit, paymentId]);
+
+      const userReqSig = optionalDeps.ethUtil.addHexPrefix(secp256k1.sign(hopDigest, userPrvBuffer).signature.toString('hex'));
+
+      return {
+        hopParams: {
+          gasPriceMax,
+          userReqSig,
+          paymentId,
+        },
+        gasLimit,
+      };
+    }).call(this).asCallback(callback);
+  }
+
+  /**
+   * Validates that the hop prebuild from the HSM is valid and correct
+   * @param wallet The wallet that the prebuild is for
+   * @param hopPrebuild The prebuild to validate
+   * @param originalParams The original parameters passed to prebuildTransaction
+   * @param callback
+   * @returns void
+   * @throws Error if The prebuild is invalid
+   */
+  validateHopPrebuild(wallet: any, hopPrebuild: HopPrebuild, originalParams: any, callback: any): void {
+    return co(function *() {
+      const {
+        tx,
+        id,
+        signature,
+      } = hopPrebuild;
+
+      const {
+        recipients,
+      } = originalParams;
+
+      // first, validate the HSM signature
+      const bitgoKeychain: { pub: string } = yield this.keychains().get({ id: wallet.keyIds()[2] });
+      const bitgoPubkeyBuffer: Buffer = prova.HDNode.fromBase58(bitgoKeychain.pub).getPublicKeyBuffer();
+      const signatureBuffer: Buffer = Buffer.from(optionalDeps.ethUtil.stripHexPrefix(signature), 'hex');
+      const messageBuffer: Buffer = Buffer.from(optionalDeps.ethUtil.stripHexPrefix(id), 'hex');
+
+      const isValidSignature: boolean = secp256k1.verify(messageBuffer, signatureBuffer.slice(1), bitgoPubkeyBuffer);
```

Comment:
Why slice the first byte off?

Reply:
its the DER signature header byte, which secp256k1.verify doesn't know about. I couldnt get bitcoinmessage (which does know about the header byte) to work, so this is how we got interoperability with the signature schemes. HSM does the same thing with my signatures